instrumentation_sql.go

```diff
@@ -430,6 +453,49 @@ func parseMySQLConnDetailsKV(connStr string) (DbConnDetails, bool) {
 	return details, true
 }
 
+// parseDB2ConnDetailsKV parses a semicolon-separated MySQL-style connection string for DB2
+func parseDB2ConnDetailsKV(connStr string) (DbConnDetails, bool) {
+	details := DbConnDetails{RawString: connStr, DatabaseName: "db2"}
+
+	for _, field := range strings.Split(connStr, ";") {
+		fieldNorm := strings.ToLower(field)
+
+		var (
+			prefix   string
+			fieldPtr *string
+		)
+		switch {
+		case strings.HasPrefix(fieldNorm, "server="):
+			address := field[len("server="):]
+			for i, v := range strings.Split(address, ":") {
+				switch i {
+				case 0:
+					details.Host = v
+				case 1:
+					details.Port = v
+				}
+			}
```

Comment:
```suggestion
			addr := strings.Split(address, ":")
			details.Host, details.Port = addr[0], addr[1]
```

Reply:
As port is optional here, the port might not be present in the slice returned by the slice.

Reply:
This code is cleaned up! Please check 😊